services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/android/controller/PpaDataRequestAndroidConverterTest.java

```diff
@@ -1,26 +1,61 @@
 
+import app.coronawarn.datadonation.common.persistence.domain.metrics.ExposureWindowTestResult;
 import app.coronawarn.datadonation.common.persistence.domain.metrics.TechnicalMetadata;
 import app.coronawarn.datadonation.common.persistence.domain.metrics.embeddable.ClientMetadataDetails;
 import app.coronawarn.datadonation.common.persistence.domain.metrics.embeddable.CwaVersionMetadata;
+import app.coronawarn.datadonation.common.persistence.service.PpaDataStorageRequest;
 import app.coronawarn.datadonation.common.protocols.internal.ppdd.ExposureRiskMetadata;
 import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPAClientMetadataAndroid;
+import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPADataAndroid;
+import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPADataIOS;
 import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPADataRequestAndroid;
+import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPADataRequestIOS;
+import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPAExposureWindow;
+import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPAExposureWindowInfectiousness;
+import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPANewExposureWindow;
 import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPARiskLevel;
 import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPASemanticVersion;
+import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPATestResult;
+import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPATestResultMetadata;
 import app.coronawarn.datadonation.common.protocols.internal.ppdd.PPAUserMetadata;
-import app.coronawarn.datadonation.services.ppac.commons.PpaDataRequestConverter;
+import app.coronawarn.datadonation.common.utils.TimeUtils;
+import app.coronawarn.datadonation.services.ppac.android.attestation.AttestationStatement;
+import app.coronawarn.datadonation.services.ppac.config.PpacConfiguration;
 import java.time.LocalDate;
 import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
 
+@ExtendWith(MockitoExtension.class)
 public class PpaDataRequestAndroidConverterTest
-    extends PpaDataRequestConverter<PPADataRequestAndroid, PPAClientMetadataAndroid> {
+    extends PpaDataRequestAndroidConverter {
+
+  private PpacConfiguration ppacConfig;
+
+  @InjectMocks
+  private AttestationStatement attestationStatement;
+
+  @BeforeEach
+  public void setup() {
+    ppacConfig = new PpacConfiguration();
+    ppacConfig.setMaxExposureWindowsToRejectSubmission(672);
```

Comment:
Should we make it clear where these magic numbers come from? I.e. extract them into a constant?

Reply:
I added a comment into the application.yaml